### ISSUE TITLE
Error when server returns status of 404

### ISSUE THREAD

**Issue by artola** [bitinn/node-fetch#128]
When making a HEAD request with 404 status code the following error occurs with node v6.2.2 and node-fetch v1.5.3:

```
events.js:160
      throw er; // Unhandled 'error' event
      ^
Error: unexpected end of file
    at Zlib._handle.onerror (zlib.js:370:17)
```

... unless pass `compress: false`

**Comment by bitinn** [comment]
Yeah, I think we ought to have a test case for where `HEAD` cause server to not send a body... as empty body is never valid gzip or deflate stream...

**Comment by bitinn** [comment]
fixed in v1.6.0
